merryface/pain_tracker
# Page routing

# Page Routing

Using Next's page routing system to be able to view different days' data. The work on this branch is not complete, but it would be worth reviewing it as it is at the moment before it gets too complex. Significant things to look at:

- A new hook has been created called `useDay`. It takes care of fetching the current day from the router
- A new util function has been created called `getDayFromUrl` which can be passed what is found in `router.query.day` and will ensure it's either a number or null.
- The footer has been updated to use `next/link`, and it's a good example to follow for other links in the site. The idea is that you make a `<Link>` component with an `href` attribute, and which contains an anchor tag, which itself has no `href`. This allows Next to preserve stated between navigation and allows it to preload pages that are linked from the current page, making things nice and fast.

## To test
Go to localhost:3000/0 and you should see the data for the first hardcoded day in state. Go to localhost:3000/1 and you should see the data for the second hardcoded day in state. Go to localhost:3000 or localhost:3000/4 or localhost:3000/whatever and you should see the default, which is the first hardcoded day in state.
